frontend/src/views/upload-project/UploadProjectViewMachine.tsx

```diff
@@ -0,0 +1,160 @@
+        unauthorized: {};
+        fileSelected: {};
+        uploading: {};
+        success: {};
+      };
+    };
+  };
+}
+export type SchemaValue = {
+  uploadProjectView: 'idle' | 'unauthorized' | 'fileSelected' | 'success';
+  toast: 'visible' | 'hidden';
+};
+
+export interface UploadProjectViewContext {
+  file: any;
+  message: string;
+  newProjectId: string;
+}
+
+export type ShowToastEvent = { type: 'SHOW_TOAST'; message: string };
+export type HideToastEvent = { type: 'HIDE_TOAST' };
+export type HandleSelectedFileEvent = { type: 'HANDLE_SELECTED_FILE'; file: string };
+export type HandleUploadEvent = { type: 'HANDLE_UPLOAD' };
+export type HandleResponseEvent = { type: 'HANDLE_RESPONSE'; response: any };
+export type UploadProjectEvent =
+  | ShowToastEvent
+  | HideToastEvent
+  | HandleSelectedFileEvent
+  | HandleUploadEvent
+  | HandleResponseEvent;
+
+export const uploadProjectMachine = Machine<UploadProjectViewContext, UploadProjectViewStateSchema, UploadProjectEvent>(
+  {
+    id: 'uploadProjectView',
+    type: 'parallel',
+    context: {
+      file: null,
+      message: null,
+      newProjectId: undefined,
+    },
+    states: {
+      toast: {
+        initial: 'hidden',
+        states: {
+          hidden: {
+            on: {
+              SHOW_TOAST: {
+                target: 'visible',
+                actions: 'setMessage',
+              },
+            },
+          },
+          visible: {
+            on: {
+              HIDE_TOAST: {
+                target: 'hidden',
+                actions: 'clearMessage',
+              },
+            },
+          },
+        },
+      },
+      uploadProjectView: {
+        initial: 'idle',
+        states: {
+          idle: {
+            on: {
+              HANDLE_SELECTED_FILE: {
+                target: 'fileSelected',
+                actions: 'setFile',
+              },
+            },
+          },
+          unauthorized: {},
+          fileSelected: {
+            on: {
+              HANDLE_UPLOAD: {
+                target: 'uploading',
+              },
+              HANDLE_SELECTED_FILE: {
+                target: 'fileSelected',
+                actions: 'setFile',
+              },
+            },
+          },
+          uploading: {
+            on: {
+              HANDLE_RESPONSE: [
+                {
+                  cond: 'isSuccess',
+                  target: 'success',
+                  actions: 'setNewProjectId',
+                },
+                {
+                  target: 'fileSelected',
+                },
+              ],
+              HANDLE_SELECTED_FILE: {
+                target: 'fileSelected',
+              },
+            },
+          },
+          success: {
+            type: 'final',
+          },
+        },
+      },
+    },
+  },
+  {
+    guards: {
+      isSuccess: (_, event) => {
+        const { response } = event as HandleResponseEvent;
+        return response?.data?.uploadProject?.project?.id;
```

Comment:
Test the response typename instead of just spamming `?` everywhere. The GraphQL API provides a clear contract on what can be null or not (hint, pretty much nothing can be null)